types/slack-mock/index.d.ts

```diff
@@ -0,0 +1,200 @@
+// Type definitions for slack-mock 1.1
+// Project: https://github.com/Skellington-Closet/slack-mock
+// Definitions by: Kris Kalavantavanich <https://github.com/kkalavantavanich>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+import * as WebSocket from "ws";
+import { IncomingHttpHeaders } from "http";
+import * as nock from "nock";
+import { Url } from "url";
+
+export = SlackMock;
+
+declare function SlackMock(config?: SlackMock.ConfigOptions): SlackMock.Instance;
+
+declare namespace SlackMock {
+    let instance: Instance;
+
+    interface Instance {
+        events: Events<any>;
```

Comment:
I couldn't say without knowing the package. `instance` is a global variable, so what determines what events are in it?

Reply:
`events` is kinda like a list where you can put things in using `send(EventUrl, T)` and get things using `calls`. The things that you put in should be the same type as the things you get out. The same goes for its siblings: `incomingWebhooks`, `interactiveButtons`, etc. From the library, I have deduced two requirements: the type should be defined by the user and the type could be different between the siblings.

The type, `T`, which I have lefted as `any`, should be defined by the user as it largely depends on which slack library the user is using. Furthermore, the type definition could change in response to an update in the slack API. 

The type may be different between siblings as it is demonstrated in the current node slack SDK.

I think that the current use of `any` could be improved so that it is more strongly typed while also compatible to usage without types in typescript. Is there anyway to achieve this?

Reply:
It looks like there is a way to get a SlackMock instance from a function instead of a global variable. Maybe the global variable should just be `let instance: Instance<unknown>` with a comment telling people to use `function SlackMock<T>(...): SlackMock.Instance<T>` instead?